File: packages/arcade_lobby/tests/components.test.ts
import { describe, it, expect } from 'vitest'
import components from '../seed/components.json'

describe('Arcade Lobby Components', () => {
  it('should be a valid array', () => {
    expect(components).toBeInstanceOf(Array)
  })

  it('should have valid component structure if components exist', () => {
    if (components.length > 0) {
      components.forEach((component: any) => {
        expect(component.id).toBeDefined()
        expect(component.type).toBeDefined()
        expect(typeof component.id).toBe('string')
        expect(typeof component.type).toBe('string')
        expect(component.children).toBeInstanceOf(Array)
      })
    }
  })
})

File: packages/archive/form_builder/seed/scripts/tests/components.test.lua
-- Component validation tests for form_builder package
-- Uses lua_test framework

describe("Form Builder Components", function()
  local components = load_cases("components.json")
  
  it("should be a valid array", function()
    expect(components).toBeType("table")
  end)
  
  it("should have valid component structure if components exist", function()
    if #components > 0 then
      for _, component in ipairs(components) do
        expect(component.id).toBeTruthy()
        expect(component.type).toBeTruthy()
        expect(component.id).toBeType("string")
        expect(component.type).toBeType("string")
      end
    end
  end)
end)
